### ISSUE TITLE
[feature request] Have a 'message sent' timestamp

### ISSUE THREAD

**Issue by yurivict** [TokTok/c-toxcore#428]
Currently offline messages show with the timestamp of arrival.
It is important for both sender and receiver to know when the message was created and sent too.

Please add the timestamp for 'send' event (when the user hits the 'send' button), and communicate it to the recipient, so that the recipient can see both times.

References:
* https://github.com/qTox/qTox/issues/3306
* https://github.com/irungentoo/toxcore/issues/1574

**Comment by iphydf** [comment]
Yes, I suggested this 2 years ago. Thanks for the issue. We'll implement it in the new protocol. Right now, it would break the application level protocol, so we'll hold off on it. This feature will be part of the new application level protocol design.

**Comment by aaannndddyyy** [comment]
+1

**Comment by isotoxin** [comment]
Isotoxin has long able to send message create timestamp.
Due to inflexible architecture of tox protocol it is impossible to implement it without break compatibility. Isotoxin sends timestamp only to another isotoxin. Oh! There is another big problem - protocol doesn't allow to determine client name to enable/disable features depend on client. Isotoxin do some tricks to detect client of contact.

**Comment by SkyzohKey** [comment]
> There is another big problem - protocol doesn't allow to determine client name to enable/disable features depend on client. 

I asked for such «client capacity detection» about 30 times, I plan to do it as a standardized custom packet in Konv.im as it'll never gets implemented.

**Comment by yurivict** [comment]
They have a plan to redesign the protocol. This must be one of the items.

**Comment by isotoxin** [comment]
> I asked for such «client capacity detection» about 30 times, I plan to do it as a standardized custom packet in Konv.im as it'll never gets implemented.

Standardized custom packet is not enough. How long client should wait such packet from other client? No. Client should know capabilities of other client just in PACKET_ID_ONLINE - first packet from other client. Isotoxin already does it by using modified core. Maybe it makes sense to add this functionality to the core officially?

**Comment by SkyzohKey** [comment]
>Maybe it makes sense to add this functionality to the core officially?

It makes sense, but wont happens i guess, as "c-toxcore" focus on tests, not features (yet).
